jolt-core/src/subprotocols/grand_product_quarks.rs

```diff
@@ -42,19 +172,20 @@ pub trait QuarkBatchedGrandProduct<C: CommitmentScheme>: Sized {
     /// Then - Constructs a g poly and preforms sumcheck proof that sum == 0
     /// Finally - computes opening proofs for a random sampled during sumcheck proof and returns
     fn prove(
-        v: Vec<DensePolynomial<C::Field>>,
+        v: &[Vec<C::Field>],
         transcript: &mut ProofTranscript,
         setup: &C::Setup,
     ) -> (Self, Vec<C::Field>);
 
     /// Verifies the given grand product proof.
+    #[allow(clippy::type_complexity)]
     fn verify(
-        self,
+        &self,
         claim: &[C::Field],
         transcript: &mut ProofTranscript,
         n_rounds: usize,
         setup: &C::Setup,
-    ) -> Result<(), QuarkError>;
+    ) -> Result<(Vec<C::Field>, Vec<C::Field>), QuarkError>;
 }
 
 impl<C: CommitmentScheme> QuarkBatchedGrandProduct<C> for QuarkGrandProductProof<C> {
```

Comment:
Instead of having the `QuarkBatchedGrandProduct` trait, let's just move these impls to the `QuarkGrandProductProof` struct (unless there's a reason not to)